File: DesignPatternsExercises/java/InterpreterDesignPattern/Solution/InterpreterDesignPattern.java
class InterpreterDesignPattern {
    public static void main(String[] args) {
    	String expressionString = "50 + 50";
		Expression expression = ExpressionReader.read(expressionString);
		System.out.println(expressionString + " = " + expression.interpret());

		expressionString = "100 - 3";
		expression = ExpressionReader.read(expressionString);
		System.out.println(expressionString + " = " + expression.interpret());

		expressionString = "5 * 5";
		expression = ExpressionReader.read(expressionString);
		System.out.println(expressionString + " = " + expression.interpret());

		expressionString = "9 / 3";
		expression = ExpressionReader.read(expressionString);
		System.out.println(expressionString + " = " + expression.interpret());
    }
}

File: DesignPatternsExercises/ruby/interpreter_design_pattern/exercise/interpreter_design_pattern.rb
require_relative 'expression_reader'

module Operators
  FIRST_CONSTANT_INDEX = 0
  SECOND_CONSTANT_INDEX = 2
  SUMMATION_OPERATOR = :+
  SUBTRACTION_OPERATOR = :-
  MULTIPLICATION_OPERATOR = :*
  DIVISION_OPERATOR = :/
end

expression_string = '50 + 50'
expression = ExpressionReader.read(expression_string)
puts "#{expression_string} = #{expression.interpret}"

expression_string = '100 - 3'
expression = ExpressionReader.read(expression_string)
puts "#{expression_string} = #{expression.interpret}"

expression_string = '5 * 5'
expression = ExpressionReader.read(expression_string);
puts "#{expression_string} = #{expression.interpret}"

expression_string = '9 / 3'
expression = ExpressionReader.read(expression_string)
puts "#{expression_string} = #{expression.interpret}"
